torch/_inductor/lowering.py

```diff
@@ -545,7 +546,12 @@ def _to_dtype(x):
 
 @register_lowering(prims.convert_element_type, type_promotion_kind=None)
 def _convert_element_type(x: TensorBox, dtype: torch.dtype):
-    if dtype.is_complex or x.get_dtype().is_complex:
+    is_complex = dtype.is_complex or x.get_dtype().is_complex
+    is_float8 = "float8" in str(dtype) or "float8" in str(x.get_dtype())
+    float8_and_unsupported = (
+        is_float8 and x.get_device().type == "cuda" and device_supports_triton_float8()
+    )
+    if is_complex or float8_and_unsupported:
```

Comment:
Should all fp8 usage fallback on unsupported hardware? If so it should probably use this:
https://github.com/pytorch/pytorch/blob/538877d2046a492a1112101e2d5d88e5754d477b/torch/_inductor/lowering.py#L1622